city_scrapers/spiders/il_port_district.py

```diff
@@ -0,0 +1,193 @@
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def parse_agendas(self, response):
+        file_names = response.xpath("//tr/td[@class='views-field views-field-title']/text()")\
+            .extract()
+        file_names = [x.strip("\n ") for x in file_names]
+        file_links = response.xpath("//tr/td/a[@class='file-download']/@href").extract()
+
+        date_pattern = r"(?<=Agenda)(.*?)(?=.pdf)"
+        agenda_dates = [re.findall(date_pattern, x)[0].replace('%20', ' ') for x in file_links]
+
+        self.files_list = []
+
+        for link, name, date in list(zip(file_links, file_names, agenda_dates)):
+            self.files_list.append({'title': name + date, 'href': link})
+
+    def parse_minutes(self, response):
+        rows = response.xpath("//tr")
+        self.minutes_dict = {}
+        for row in rows:
+            file_name = row.xpath(".//td[@class='views-field views-field-title']/text()")\
+                .extract_first()
+            if not file_name:
+                continue
+
+            file_name = file_name.strip("\n ")
+            file_name_dt = re.findall(r"(?:.*?)(?:\d{4})", file_name)[0]
+            file_name_dt = datetime.datetime.strptime(file_name_dt, "%B %d, %Y")
+
+            file_link = row.xpath(".//td[@class='views-field views-field-field-file']/a/@href")\
+                .extract_first()
+
+            self.minutes_dict.setdefault(file_name_dt.date(), file_link)
+
+    def _parse_classification(self, i, meeting_types):
+        if "board" in meeting_types.lower():
+            return BOARD
+        elif "committee" in meeting_types.lower():
+            return COMMITTEE
+        else:
+            return NOT_CLASSIFIED
+
+    def _parse_start(self, date, year, meeting_time):
+
+        if date.startswith("***") or date.endswith("***"):
+            meeting_time = self.changed_meeting_time
+        elif date.startswith("**") or date.endswith("**"):
+            self.special_meeting = "Special "
+        elif date.startswith("*") or date.endswith("*"):
+            new_date = re.findall(r"(?<=\()(.*?)(?=NEW)", date)
+            date = new_date[0][:-3] if new_date else date
+
+        date = date.strip(" *")
+        dt = " ".join([year, date, meeting_time])
+        dt = datetime.datetime.strptime(dt, "%Y %B %d %I:%M%p")
+
+        return dt
+
+    def _parse_agendas_links(self, classification, start):
+        date = datetime.datetime.strftime(start, '%B %Y')
+        if classification == 'Board':
+            link_list = [d for d in self.files_list if 'Board' in d['title'] and date in d['title']]
+        elif classification == 'Committee':
+            link_list = [
+                d for d in self.files_list if 'Committee' in d['title'] and date in d['title']
+            ]
+        else:
+            link_list = []
+
+        return link_list
+
+    def _parse_minutes_links(self, start):
+        link = self.minutes_dict.get(start.date(), None)
```

Comment:
This could be simplified by making the values in `minutes_dict` lists with the title already applied so that this method wouldn't be needed and you could just call `self.minutes_dict.get(start.date(), [])`